cloudconfig/podcfg/image.go

```diff
@@ -42,44 +42,40 @@ func IsJujuOCIImage(imagePath string) bool {
 }
 
 // GetJujuOCIImagePath returns the jujud oci image path.
-func GetJujuOCIImagePath(controllerCfg controller.Config, ver version.Number, build int) string {
+func GetJujuOCIImagePath(controllerCfg controller.Config, ver version.Number, build int) (string, error) {
 	// First check the deprecated "caas-operator-image-path" config.
 	ver.Build = build
-	imagePath := RebuildOldOperatorImagePath(
+	imagePath, err := RebuildOldOperatorImagePath(
 		controllerCfg.CAASOperatorImagePath(), ver,
 	)
-	if imagePath != "" {
-		return imagePath
+	if imagePath != "" || err != nil {
+		return imagePath, err
 	}
 	return imageRepoToPath(controllerCfg.CAASImageRepo(), ver)
 }
 
 // RebuildOldOperatorImagePath returns a updated image path for the specified juju version.
-func RebuildOldOperatorImagePath(imagePath string, ver version.Number) string {
+func RebuildOldOperatorImagePath(imagePath string, ver version.Number) (string, error) {
 	if imagePath == "" {
-		return ""
+		return "", nil
 	}
 	return tagImagePath(imagePath, ver)
 }
 
-func tagImagePath(path string, ver version.Number) string {
-	var verString string
-	splittedPath := strings.Split(path, ":")
-	path = splittedPath[0]
-	if len(splittedPath) > 1 {
-		verString = splittedPath[1]
+func tagImagePath(fullPath string, ver version.Number) (string, error) {
+	ref, err := reference.Parse(fullPath)
+	if err != nil {
+		return "", errors.Trace(err)
 	}
+	imageNamed := ref.(reference.Named)
```

Comment:
this could panic, but probably ok if it did. (e.g. digest only path isn't a reference.Named)

Reply:
i thought about returning an error here but checked the call sites and we are good. maybe i should return an error anyway just as a safety check